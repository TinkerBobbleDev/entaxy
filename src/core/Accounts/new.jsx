import React from 'react'
import { connect } from 'react-redux'
import PropTypes from 'prop-types'
import AccountForm from './form'
import { createAccount } from '../../store/accounts/actions'

const mapDispatchToProps = {
  handleSave: account => createAccount(account)
}

export const NewAccountComponent = ({ history, handleSave }) => {
  const onSave = (account) => {
    handleSave(account).then((accountId) => {
      history.push(`/accounts/${accountId}/transactions`)
    })
  }

  const onCancel = () => {
    history.push('/dashboard')
  }

  return (
    <AccountForm handleSave={onSave} handleCancel={onCancel} />
  )
}

NewAccountComponent.propTypes = {
  history: PropTypes.object.isRequired,
  handleSave: PropTypes.func.isRequired
}

export default connect(null, mapDispatchToProps)(NewAccountComponent)
