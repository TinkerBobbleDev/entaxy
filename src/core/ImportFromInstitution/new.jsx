import React from 'react'
import { connect } from 'react-redux'
import PropTypes from 'prop-types'
import ImportFromInstitutionForm from './form'
import { createAccountGroup } from '../../store/accounts/actions'

const mapDispatchToProps = {
  createAccountGroup: (institution, accountGroup, accounts) =>
    createAccountGroup(institution, accountGroup, accounts)
}

export class NewImportFromInstitutionComponent extends React.Component {
  handleSave = (accountGroup, accounts) => {
    const { institution } = this.props.match.params
    this.props.createAccountGroup(institution, accountGroup, accounts)
    this.props.history.push('/dashboard')
  }

  handleCancel = () => {
    this.props.history.push('/dashboard')
  }

  render() {
    const { institution } = this.props.match.params
    return (
      <ImportFromInstitutionForm
        handleSave={this.handleSave}
        handleCancel={this.handleCancel}
        institution={institution}
      />
    )
  }
}

NewImportFromInstitutionComponent.propTypes = {
  history: PropTypes.object.isRequired,
  createAccountGroup: PropTypes.func.isRequired,
  match: PropTypes.object.isRequired
}

export default connect(null, mapDispatchToProps)(NewImportFromInstitutionComponent)
