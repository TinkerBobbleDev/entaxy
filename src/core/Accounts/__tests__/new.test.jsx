import React from 'react'
import renderer from 'react-test-renderer'
import { NewAccountComponent } from '../new'

jest.mock('../form', () => 'AccountForm')

describe('New Account', () => {
  const mochHandleSave = jest.fn()
  const mochHandleCancel = jest.fn()

  beforeEach(() => {
    jest.clearAllMocks()
  })

  it('matches snapshot', () => {
    const component = renderer.create((
      <NewAccountComponent
        handleSave={mochHandleSave}
        handleCancel={mochHandleCancel}
        history={{}}
      />
    ))
    expect(component.toJSON()).toMatchSnapshot()
  })
})
