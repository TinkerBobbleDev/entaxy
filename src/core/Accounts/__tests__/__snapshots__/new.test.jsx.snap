// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`New Account matches snapshot 1`] = `
<AccountForm
  handleCancel={[Function]}
  handleSave={[Function]}
/>
`;
